implemented translation feature

Shinigami92 opened the pull request:
next steps:
* german locales
* improve selection of language

Shinigami92 committed:
Added angular translate
english HALLOWEEN_2016 complete
german as second language -> test with Hexe
Added language selection-box to settings (needs improvement :( )

Shinigami92 committed:
fix conflicts

Shinigami92 committed:
Merge branch 'Js41637-master'

Shinigami92 committed:
Added generator for locales

Sort langKeys

Shinigami92 committed:
implemented translation for items

Shinigami92 committed:
Fix conflicts

Shinigami92 committed:
Merge branch 'Js41637-master'

Shinigami92 committed:
Improved language-selection

The side doesn't reload anymore after you selected a language :)

Shinigami92 commented:
improved language selection -> DONE

Shinigami92 committed:
Added german YEAR_OF_THE_ROOSTER_2017 translation

Shinigami92 commented:
german translation for current event added

Js41637 commented:
Hey thanks for this, I won't be able to merge now as we're in the process of a lot of core changes to the layout and backend. I will merge it into a branch later and make a few changes for how I would like it.

Shinigami92 commented:
Ok so now I go playing a bit overwatch
But can you tell me, whether you can extract the language names from an API? So I don't have to write everything manualy?
Otherwise this is lot of work... but makeable

Js41637 commented:
I'm sure you can just run it all through google translate, I won't be manually translating anything, you don't have to translate it all if you don't want too, i'm sure if someone wants it enough they can do it.

Shinigami92 commented:
OK, then I will translate everything manually to german with my game
if someone want france or anything else, s/he can do it :)

Shinigami92 committed:
Merge pull request #3 from Js41637/master

merge master to fork

Shinigami92 committed:
Added german translation for WINTER_WONDERLAND_2016

Shinigami92 commented:
Link to issue: [issue#24](https://github.com/Js41637/Overwatch-Item-Tracker/issues/24)

Shinigami92 committed:
Added german translation for HALLOWEEN_2016

Shinigami92 committed:
Removed '::' infront of name-variables in views
cause this breakes live-language translation

Shinigami92 committed:
Merge branch 'master' into master

Shinigami92 committed:
Merge pull request #4 from Js41637/master

merge master to fork

Shinigami92 committed:
Add german translation for SUMMER_GAMES_2016

JigAma commented:
I've started to implement French translation locally.

JigAma commented:
I finished the French localization but i can't push my changes because this push request hasn't been merged yet

Shinigami92 commented:
@JigAma I have invited you to my fork.
Now you can make a new branch on my fork, commit your french localization to it and the make a merge request to my master-branch in my fork
I will review the changes you made and the it will be added in this pull request

nduriez committed:
Added French Translation

Added "fr_FR" to $rootScope.languages in js\main.js
Created the file "fr_FR.json" in \locales

Shinigami92 committed:
Merge pull request #5 from Shinigami92/French-Translation

Added French Translation

Shinigami92 committed:
Fixed " and , in branch from JigAma

Shinigami92 commented:
Review french translation done

Shinigami92 committed:
Merge master into fork and fix conflicts

Shinigami92 committed:
Merge branch 'Js41637-master'

Shinigami92 committed:
fix translations for new theme

fix flags in settings menu

Shinigami92 committed:
Revert to static binding

Make Js41637 happy again

Shinigami92 committed:
fix merge conflicts

Shinigami92 committed:
Merge branch 'Js41637-master'

Shinigami92 committed:
Merge branch 'master' into master

Shinigami92 commented:
I will soon look at all the changes and then continue with the translation.
But maybe first I will wait a bit until the master isn't rapidly updated anymore :)

Shinigami92 committed:
Resolve conflicts and updated to newest master after new-thing branch

Improved language file generation

Shinigami92 committed:
Merge branch 'Js41637-master'

Shinigami92 commented:
translation from language-keys to english is working
de and fr are only working for events, because the old translations were compatible. only some aren't and will fallback to english

so i will translate the other missing keys the next days
and hopefully JigAma will be back and translate the french

Shinigami92 committed:
Fixed differences from old to new version for locales/de_DE

Shinigami92 committed:
Added Uprising 2017 locales/de_DE

Shinigami92 committed:
Fixed some halloween translations for locales/de_DE

Shinigami92 committed:
Fixed hero.html grid item.name not translated

Shinigami92 committed:
Added german all.icons

Shinigami92 commented:
Sorry that I'm working a bit slow
Hav currently some exams

JigAma commented:
I'm waiting Js to merge the branch as it is right now before making the new translation because we hadn't have news since a bit of time now and I don't want to do it if i'm not sure it will be used

Shinigami92 commented:
Yes me too
I havn't any response from JS whether he want to merge this branch 
Currently i have translated all event items but not the regular class items (they will fallback to english) 
It is a huge amount of work to translate the items and there are dublicates in event and regular with the current datastructur

Shinigami92 committed:
Merge branch 'master' into master

Shinigami92 committed:
recreated /locales/en_US.json with /locales/language-generator.html

Shinigami92 committed:
added translation to anniversaey_2017 sprays
